binding/Binding/SKObject.cs

```diff
@@ -143,22 +143,20 @@ internal static void DeregisterHandle (IntPtr handle, SKObject instance)
 			if (handle == IntPtr.Zero)
 				return;
 
-			var nonExistent = new WeakReference (null);
-			var weak = instances.AddOrUpdate (handle, nonExistent, (ptr, old) => {
-				if (old.Target is SKObject obj && !obj.IsDisposed)
-					return old;
-				return new WeakReference (null);
-			});
+			var existed = instances.TryRemove (handle, out var weak);
+			if (existed && weak.Target is SKObject obj && !obj.IsDisposed) {
+				// Existing object is not disposed, so re-register it
+				RegisterHandle (handle, obj);
```

Comment:
We need to be careful here with concurrency. The reason I used `AddOrUpdate` was because the key was "locked" during this block so I could do everything.

With two separate statements, the object could be valid, then while starting to execute the `RegisterHandle` the object could be disposed. We would then be registering a `null` handle. (but, looking at the code for `RegisterHandle`, this may not actually be a problem because we have some checks and registering a `null` handle may not be such an issue)

I had a big, fat `lock` around the register/deregister logic before and that was nice to prevent any concurrency, but it was overkill.

(I wrote a test that maybe needs a bit more concurrency testing: https://github.com/mono/SkiaSharp/blob/v1.68.1/tests/Tests/SKObjectTest.cs#L267-L307)

Just a thought. I think this is good to merge, but just wanted to see if you had any further comments.

Reply:
Makes sense. Because of these concerns, I used `RegisterHandle` instead of more logic in this method. However, I am unsure if it is sufficient in every concurrent scenario.

Reply:
Gillibald mentioned "In the register method you just do the same check for IsDisposed == true and do nothing in that case."

Should something like that be part of this PR? If so, would that just be part of the guard clause, or would `RegisterHandle(...).Update(...)` need to be modified as well?

Reply:
I think this PR is good.

Reply:
Cool deal, thanks 😎